### ISSUE TITLE
Manual helm v2 install without Tiller on EKS - unknown field "affinity"

### ISSUE THREAD

**Issue by bliu-rei** [lucidworks/fusion-cloud-native#17]
Following your instructions under **Use Helm to Install Fusion**, subsection **Don't want to use Tiller? (Helm v2)**, on to AWS EKS cluster:

- For LoadBalancer service type, [service annotation is required for EKS](https://docs.aws.amazon.com/eks/latest/userguide/load-balancing.html)
  - `service.beta.kubernetes.io/aws-load-balancer-type: nlb` for public facing
  - `service.beta.kubernetes.io/aws-load-balancer-internal: 0.0.0.0/0` for private facing
  so we must use customization values.yaml. In our case, EKS is private only, so the customized values.yaml content is as follows:

```
sql-service:
  service:
    thrift:
      type: "ClusterIP"
api-gateway:
  service:
    annotations: { service.beta.kubernetes.io/aws-load-balancer-internal: 0.0.0.0/0 }
```

With the above, and with RELEASE and NAMESPACE both set to 'f5', run `kubectl apply -f "${RELEASE}_${NAMESPACE}_fusion_install.yaml" --namespace "${NAMESPACE}"`, got validation error:
```
error: error validating "f5_f5_fusion_install.yaml": error validating data: ValidationError(Deployment.spec): unknown field "affinity" in io.k8s.api.apps.v1.DeploymentSpec; if you choose to ignore these errors, turn validation off with --validate=false
```

Rerun kubectl with `--validate=false`, deployment succeeds, and http://<internal-load-balancer-of-api-gateway-service>:6764 from Chrome brings up a log in page!

**Comment by bliu-rei** [comment]
If use `helm install --timeout 240...` (with Tiller configured and verified to be working), got the following error:
```
Error: no kind "Job" is registered for version "batch/v1" in scheme "k8s.io/kubernetes/pkg/api/legacyscheme/scheme.go:30"
```
Even if the above error can be resolved somehow, I would expect the "affinity" error will come up next

**Comment by bliu-rei** [comment]
![Capture](https://user-images.githubusercontent.com/55470908/68418108-775dce00-014c-11ea-9830-2d5140f8cceb.PNG)

That was the page shown in Chrome to confirm deployment success

**Comment by bliu-rei** [comment]
Here's the output from `kubectl get all -n f5`
```
NAME                                         READY   STATUS      RESTARTS   AGE
pod/f5-admin-ui-6984fbfd64-6wqwn             1/1     Running     0          101m
pod/f5-api-gateway-76bb564656-2k5d9          1/1     Running     0          101m
pod/f5-api-gateway-fpx2h                     0/1     Completed   0          144m
pod/f5-auth-ui-6fcc49f868-9bqtc              1/1     Running     0          101m
pod/f5-classic-rest-service-0                1/1     Running     1          100m
pod/f5-cx-api-0                              1/1     Running     0          98m
pod/f5-cx-scheduler-6fbc4f5b5c-p7kwp         1/1     Running     0          100m
pod/f5-cx-script-executor-57f7889b84-7hhq2   1/1     Running     0          100m
pod/f5-cx-ui-688849cb86-d4cvh                1/1     Running     0          98m
pod/f5-cx-ui-688849cb86-d6fgf                1/1     Running     0          98m
pod/f5-cx-user-prefs-548899ff9f-bzs7q        1/1     Running     0          98m
pod/f5-devops-ui-9c8c9bc94-djg6b             1/1     Running     0          101m
pod/f5-fusion-admin-6888d66579-rkjxs         1/1     Running     1          101m
pod/f5-fusion-indexing-6cb8c9cf78-fjzck      1/1     Running     1          101m
pod/f5-insights-7bb564c554-csjqn             1/1     Running     0          101m
pod/f5-job-launcher-749dc498c6-v2zbk         1/1     Running     0          101m
pod/f5-job-rest-server-7d574c857-6chqs       1/1     Running     1          101m
pod/f5-kafka-0                               1/1     Running     1          100m
pod/f5-kafka-1                               1/1     Running     0          99m
pod/f5-kafka-2                               1/1     Running     0          98m
pod/f5-logstash-0                            1/1     Running     3          100m
pod/f5-ml-model-service-64964d68d7-hnvdh     2/2     Running     0          100m
pod/f5-query-pipeline-8656b94784-hjhgw       1/1     Running     0          100m
pod/f5-rest-service-56544dff5f-4nxvb         1/1     Running     0          100m
pod/f5-rpc-service-859854469-4hzjz           1/1     Running     1          100m
pod/f5-rules-ui-79866ffb98-c8z45             1/1     Running     0          100m
pod/f5-solr-0                                1/1     Running     0          100m
pod/f5-solr-1                                1/1     Running     0          99m
pod/f5-solr-2                                1/1     Running     0          98m
pod/f5-sql-service-cm-6d4d5bd44-j74rn        1/1     Running     1          100m
pod/f5-sql-service-cr-869bfdf675-qzjq5       0/1     Pending     0          100m
pod/f5-webapps-5b564885b5-cpnc8              1/1     Running     0          100m
pod/f5-zookeeper-0                           1/1     Running     0          100m
pod/f5-zookeeper-1                           1/1     Running     0          100m
pod/f5-zookeeper-2                           1/1     Running     0          99m

NAME                                  TYPE           CLUSTER-IP       EXTERNAL-IP                                                                       PORT(S)                      AGE
service/admin                         ClusterIP      172.20.138.216   <none>                                                                            8765/TCP                     101m
service/admin-ui                      ClusterIP      172.20.248.92    <none>                                                                            8080/TCP                     101m
service/auth-ui                       ClusterIP      172.20.71.136    <none>                                                                            8080/TCP                     101m
service/connectors                    ClusterIP      172.20.150.243   <none>                                                                            9010/TCP                     101m
service/connectors-classic            ClusterIP      172.20.226.229   <none>                                                                            9000/TCP                     101m
service/connectors-rpc                ClusterIP      172.20.210.145   <none>                                                                            8771/TCP                     101m
service/cx-api                        ClusterIP      172.20.74.130    <none>                                                                            80/TCP                       101m
service/cx-api-headless               ClusterIP      None             <none>                                                                            80/TCP                       101m
service/cx-scheduler                  ClusterIP      172.20.207.112   <none>                                                                            80/TCP                       101m
service/cx-scheduler-headless         ClusterIP      None             <none>                                                                            80/TCP                       101m
service/cx-script-executor            ClusterIP      172.20.77.152    <none>                                                                            80/TCP                       101m
service/cx-script-executor-headless   ClusterIP      None             <none>                                                                            80/TCP                       101m
service/cx-ui                         ClusterIP      172.20.51.173    <none>                                                                            80/TCP                       98m
service/devops-ui                     ClusterIP      172.20.16.198    <none>                                                                            8080/TCP                     101m
service/f5-cx-user-prefs              ClusterIP      172.20.201.130   <none>                                                                            80/TCP                       101m
service/f5-cx-user-prefs-headless     ClusterIP      None             <none>                                                                            80/TCP                       101m
service/f5-kafka                      ClusterIP      172.20.242.168   <none>                                                                            9092/TCP                     101m
service/f5-kafka-headless             ClusterIP      None             <none>                                                                            9092/TCP                     101m
service/f5-logstash                   ClusterIP      172.20.26.20     <none>                                                                            5044/TCP,9600/TCP,4560/TCP   101m
service/f5-solr-headless              ClusterIP      None             <none>                                                                            8983/TCP                     98m
service/f5-solr-svc                   ClusterIP      172.20.137.71    <none>                                                                            8983/TCP                     98m
service/f5-zookeeper                  ClusterIP      172.20.65.144    <none>                                                                            2181/TCP                     101m
service/f5-zookeeper-headless         ClusterIP      None             <none>                                                                            2181/TCP,3888/TCP,2888/TCP   101m
service/indexing                      ClusterIP      172.20.118.65    <none>                                                                            8765/TCP                     101m
service/insights                      ClusterIP      172.20.252.108   <none>                                                                            8080/TCP                     101m
service/job-launcher                  ClusterIP      172.20.79.41     <none>                                                                            8083/TCP                     101m
service/job-rest-server               ClusterIP      172.20.22.122    <none>                                                                            8081/TCP                     101m
service/ml-model-grpc                 ClusterIP      172.20.72.179    <none>                                                                            6565/TCP                     101m
service/ml-model-service              ClusterIP      172.20.134.79    <none>                                                                            8086/TCP                     101m
service/proxy                         LoadBalancer   172.20.119.245   internal-ad14e6930018111eab1970a869e58dbd-487557529.us-west-2.elb.amazonaws.com   6764:31266/TCP               101m
service/query                         ClusterIP      172.20.131.9     <none>                                                                            8787/TCP                     101m
service/rules-ui                      ClusterIP      172.20.241.45    <none>                                                                            8080/TCP                     101m
service/sql-jdbc                      ClusterIP      172.20.54.64     <none>                                                                            8088/TCP                     101m
service/sql-service                   ClusterIP      172.20.116.117   <none>                                                                            8085/TCP                     101m
service/webapps                       ClusterIP      172.20.76.93     <none>                                                                            8780/TCP                     101m

NAME                                    READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/f5-admin-ui             1/1     1            1           101m
deployment.apps/f5-api-gateway          1/1     1            1           101m
deployment.apps/f5-auth-ui              1/1     1            1           101m
deployment.apps/f5-cx-scheduler         1/1     1            1           100m
deployment.apps/f5-cx-script-executor   1/1     1            1           100m
deployment.apps/f5-cx-ui                2/2     2            2           98m
deployment.apps/f5-cx-user-prefs        1/1     1            1           98m
deployment.apps/f5-devops-ui            1/1     1            1           101m
deployment.apps/f5-fusion-admin         1/1     1            1           101m
deployment.apps/f5-fusion-indexing      1/1     1            1           101m
deployment.apps/f5-insights             1/1     1            1           101m
deployment.apps/f5-job-launcher         1/1     1            1           101m
deployment.apps/f5-job-rest-server      1/1     1            1           101m
deployment.apps/f5-ml-model-service     1/1     1            1           101m
deployment.apps/f5-query-pipeline       1/1     1            1           100m
deployment.apps/f5-rest-service         1/1     1            1           100m
deployment.apps/f5-rpc-service          1/1     1            1           100m
deployment.apps/f5-rules-ui             1/1     1            1           100m
deployment.apps/f5-sql-service-cm       1/1     1            1           100m
deployment.apps/f5-sql-service-cr       0/1     1            0           100m
deployment.apps/f5-webapps              1/1     1            1           100m

NAME                                               DESIRED   CURRENT   READY   AGE
replicaset.apps/f5-admin-ui-6984fbfd64             1         1         1       101m
replicaset.apps/f5-api-gateway-76bb564656          1         1         1       101m
replicaset.apps/f5-auth-ui-6fcc49f868              1         1         1       101m
replicaset.apps/f5-cx-scheduler-6fbc4f5b5c         1         1         1       100m
replicaset.apps/f5-cx-script-executor-57f7889b84   1         1         1       100m
replicaset.apps/f5-cx-ui-688849cb86                2         2         2       98m
replicaset.apps/f5-cx-user-prefs-548899ff9f        1         1         1       98m
replicaset.apps/f5-devops-ui-9c8c9bc94             1         1         1       101m
replicaset.apps/f5-fusion-admin-6888d66579         1         1         1       101m
replicaset.apps/f5-fusion-indexing-6cb8c9cf78      1         1         1       101m
replicaset.apps/f5-insights-7bb564c554             1         1         1       101m
replicaset.apps/f5-job-launcher-749dc498c6         1         1         1       101m
replicaset.apps/f5-job-rest-server-7d574c857       1         1         1       101m
replicaset.apps/f5-ml-model-service-64964d68d7     1         1         1       100m
replicaset.apps/f5-query-pipeline-8656b94784       1         1         1       100m
replicaset.apps/f5-rest-service-56544dff5f         1         1         1       100m
replicaset.apps/f5-rpc-service-859854469           1         1         1       100m
replicaset.apps/f5-rules-ui-79866ffb98             1         1         1       100m
replicaset.apps/f5-sql-service-cm-6d4d5bd44        1         1         1       100m
replicaset.apps/f5-sql-service-cr-869bfdf675       1         1         0       100m
replicaset.apps/f5-webapps-5b564885b5              1         1         1       100m

NAME                                       READY   AGE
statefulset.apps/f5-classic-rest-service   1/1     100m
statefulset.apps/f5-cx-api                 1/1     98m
statefulset.apps/f5-kafka                  3/3     100m
statefulset.apps/f5-logstash               1/1     100m
statefulset.apps/f5-solr                   3/3     100m
statefulset.apps/f5-zookeeper              3/3     100m

NAME                                           REFERENCE             TARGETS         MINPODS   MAXPODS   REPLICAS   AGE
horizontalpodautoscaler.autoscaling/f5-cx-ui   Deployment/f5-cx-ui   <unknown>/70%   2         10        2          100m

NAME                       COMPLETIONS   DURATION   AGE
job.batch/f5-api-gateway   1/1           3s         144m

NAME                                          SCHEDULE       SUSPEND   ACTIVE   LAST SCHEDULE   AGE
cronjob.batch/f5-job-launcher-spark-cleanup   */15 * * * *   False     0        8m49s           100m
```

**Comment by thelabdude** [comment]
```
Error: no kind "Job" is registered for version "batch/v1" in scheme "k8s.io/kubernetes/pkg/api/legacyscheme/scheme.go:30"
```
Is a bug in Helm v2.16.0, see: https://github.com/helm/helm/issues/6894

Maybe switch to using Helm v3?

**Comment by thelabdude** [comment]
Pull the latest changes in this repo too ... should fix the affinity issue you saw.

**Comment by bliu-rei** [comment]
Thanks for your response. 

We can ignore `Error: no kind "Job" is registered for version "batch/v1" in scheme "k8s.io/kubernetes/pkg/api/legacyscheme/scheme.go:30"` for now, because we don't intend to use `helm install`. Instead we will use `kubectl apply` on the yaml file generated by `helm template`

For the **affinity** issue itself, I'm able to trace it down to `charts\cx-ui\templates\deployment.yaml` that was generating this error. [Pull request #5](https://github.com/lucidworks/fusion-cloud-native/pull/5) does not seem to address the issue, or providing an example on how.

**Comment by ian-thebridge-lucidworks** [comment]
Hi @bliu-rei, The problem with the `affinity` field was fixed in an update to the helm chart. If you regenerate the `yaml` file that you are applying with the command:

```
helm fetch --untar lucidworks/fusion --version 5.0.2-6
```

and then execute the remaining commands to not use tiller:

```
helm template -n "${RELEASE}" --namespace "${NAMESPACE}" --values "${MY_VALUES}" fusion > ${RELEASE}_${NAMESPACE}_fusion_install.yaml
kubectl apply -f "${RELEASE}_${NAMESPACE}_fusion_install.yaml" --namespace "${NAMESPACE}"
```
This should then apply cleanly. Can you try this and let us know if it works?

**Comment by bliu-rei** [comment]
Hi Ian,

I can confirm that the `affinity` issue is fixed in **fusion 5.0.2-6** chart -- I'm able to run `kubectl  apply` successfully without any error on the generated yaml from **fusion 5.0.2-6**. 

Thank you very much!
